p4-16/psa/PSA.mdk

```diff
@@ -199,6 +202,41 @@ copy going to the control CPU to have a software-defined header
 (defined in the P4 program) that is different than any headers on the
 packet copies going to the Egress control block.
 
+A PSA implementation may choose to implement multiple class-of-service
+queues for each output port, but it may also choose to implement only
+1 queue for each output port.  The Ingress control block may choose to
```

Comment:
Are mandating VOQ here as a principle? I do not think that's the intention. Perhaps we can add "or use any other queuing arrangement".

Reply:
Are you thinking that a typical PSA implementation would have 1 separate FIFO queue per output port, or less than that?  If less than that, you can have significant head of line blocking for traffic leading to individual output ports.

If we take 1 FIFO queue per output port as a minimum, then I was thinking that if a PSA implementation supported 4 class of service queues, it would be 4 class of service queues per output port.  Anything less and you are introducing head of line blocking issues.

But I could certainly be bringing in unwarranted assumptions here that are best gotten out in the open before finishing the PSA spec.

Reply:
As an example, there can be a separate queue or queues for multicast packets, etc, but ultimately, it is not relevant to this very simple discussion that we are allowing to have this field that and that's pretty much the only thing I would say. We are not going to define its semantics further.

Reply:
I have changed the text not to mention the queueing structure, except that a PSA implementation might implement different FIFO queues for different classes of service.